daft/delta_lake/delta_lake_scan.py

```diff
@@ -0,0 +1,187 @@
+from __future__ import annotations
+
+import logging
+import os
+from collections.abc import Iterator
+from typing import Any
+from urllib.parse import urlparse
+
+import pyarrow as pa
+from deltalake.table import DeltaTable
+
+import daft
+from daft.daft import (
+    AzureConfig,
+    FileFormatConfig,
+    GCSConfig,
+    NativeStorageConfig,
+    ParquetSourceConfig,
+    Pushdowns,
+    S3Config,
+    ScanTask,
+    StorageConfig,
+)
+from daft.io.scan import PartitionField, ScanOperator
+from daft.logical.schema import Schema
+
+logger = logging.getLogger(__name__)
+
+
+class DeltaLakeScanOperator(ScanOperator):
+    def __init__(self, table_uri: str, storage_config: StorageConfig) -> None:
+        super().__init__()
+        storage_options = _storage_config_to_storage_options(storage_config, table_uri)
+        self._table = DeltaTable(table_uri, storage_options=storage_options)
+        self._storage_config = storage_config
+        self._schema = Schema.from_pyarrow_schema(self._table.schema().to_pyarrow())
+        partition_columns = set(self._table.metadata().partition_columns)
+        self._partition_keys = [
+            PartitionField(field._field) for field in self._schema if field.name in partition_columns
+        ]
+
+    def schema(self) -> Schema:
+        return self._schema
+
+    def display_name(self) -> str:
+        return f"DeltaLakeScanOperator({self._table.metadata().name})"
+
+    def partitioning_keys(self) -> list[PartitionField]:
+        return self._partition_keys
+
+    def multiline_display(self) -> list[str]:
+        return [
+            self.display_name(),
+            f"Schema = {self._schema}",
+            f"Partitioning keys = {self.partitioning_keys()}",
+            # TODO(Clark): Improve repr of storage config here.
+            f"Storage config = {self._storage_config}",
+        ]
+
+    def to_scan_tasks(self, pushdowns: Pushdowns) -> Iterator[ScanTask]:
+        # TODO(Clark): Push limit and filter expressions into deltalake action fetch, to prune the files returned.
+        paths = self._table.file_uris()
```

Comment:
Is there a reason we can't use the paths from `add_actions` and have to use `self._table.file_uris()`?

Reply:
Yep, the paths in `add_actions` only consists of the file name, rather than the full path/URI.

Reply:
cool, got it